### ISSUE TITLE
Wrong response body for new endpoint '/api/v2/delete'

### ISSUE THREAD

**Issue by lnshi** [cayleygraph/cayley#530]
```
{
  "result": "Successfully wrote 11 quads.",
  "count": 11
}
```

Supposed to be **deleted**, haha

**Comment by lnshi** [comment]
And @dennwc 

After the deletion, I checked the `nodes` collection in MongoDB, it is empty, and it is quite reasonable, but still have bunch of data in the `quads` collection, can u briefly explain a bit the design here?

**Comment by lnshi** [comment]
And @dennwc 

Sorry, one more question regarding the deleteion, as u can check here [test_purpose.nq](https://github.com/lnshi/node-cayley/blob/master/test/data/test_purpose.nq), same to below content:
```
</user/shortid/23TplPdS> <userId> 23TplPdS companyA .
</user/shortid/23TplPdS> <userSetId> XXX_L2 companyA .
</user/shortid/23TplPdS> <realName> XXX_L3 companyA .
</user/shortid/23TplPdS> <displayName> XXX_L4 companyA .
</user/shortid/23TplPdS> <gender> M companyA .
</user/shortid/23TplPdS> <email> xxx.l6@xxx.com companyA .
</user/shortid/23TplPdS> <isEmailVerified> false companyA .
</user/shortid/23TplPdS> <mobilePhone> _:BN@</user/shortid/23TplPdS>.mobilePhone companyA .
_:BN@</user/shortid/23TplPdS>.mobilePhone <isVerified> false companyA .
_:BN@</user/shortid/23TplPdS>.mobilePhone <alpha3CountryCode> +65 companyA .
_:BN@</user/shortid/23TplPdS>.mobilePhone <mobilePhoneNoWithCountryCallingCode> +6586720011 companyA .
```

In the nq file if I use the single quote `'` or double quote `"` for the `Object` values, like `23TplPdS` or `Lable` values, like `companyA`, whatever I do I cannot use the delete API to successfully delete the data, but If I just leave it as plain bare value, then I will be able to delete all successfully, u can refer to the test case here: [/v1/write/file/nquad line:224](https://github.com/lnshi/node-cayley/blob/master/test/apis/callback.js), so two concerns here:

1. How Cayley deal with the single quote `'` and double quote `"` in the `Object` and 'Label' fields.
2. If user must use plain bare value, then how to deal with the condition that the value iteself contains space.

**Comment by dennwc** [comment]
@lnshi Regarding Mongo, it might be an append-only log of database operations. In any case we will rewrite Mongo code for reification.

@lnshi Regarding your nquads example, I don't think it's written according to standard. Cayley may accept it now, but it might stop doing so eventually :) In particular, all object values should be quoted and escaped. Currently `true` is interpretted as raw byte array in Cayley, and not string `true` which you would expect. The correct value should be `"true"`. Same for Label.

And for writing structured data, I'd recommend using JSON-LD format instead of nquads. Don't forget to set `format` parameter or `Content-Type` header in this case, so Cayley can determine format properly.

**Comment by lnshi** [comment]
@dennwc thank u for the response.

Yes, I know all the `object` in N-Quads should be quoted with `"`, as u can see all other test cases based on the test data here [friend_circle_with_label.nq](https://github.com/lnshi/node-cayley/blob/master/test/data/friend_circle_with_label.nq), I believe they are all proper N-Quads data.

@dennwc After compared the data wrote in by `/api/v2/write` and `/api/v1/write/file/nquad`, I think I found out the problem, **they are not compatible**, so: `/api/v2/delete` can only delete the data which are written in by `/api/v2/write`, and `/api/v1/delete` can only delete the data which are written in by `/api/v1/write` `/api/v1/write/file/nquad`, I just adjusted the test cases to use the same API version. Any plan regarding the HTTP APIs incompatibleness or just leave it there?

@dennwc  Now cayley already has the APIs for writing, reading, parsing, deleting JSON-LD data?

**Comment by lnshi** [comment]
@dennwc Try to use the `/api/v2/delete` to delete the data which are written by `/api/v1/write/file/nquad`, then u will be able to experience what I was encountering.

**Comment by lnshi** [comment]
@dennwc  try to use `/api/v2/delete` to delete the data which are written in by `/api/v1/write/file/nquad` then u will be able to experience what I was encountering.

**Comment by dennwc** [comment]
@lnshi Checked `/api/v2/delete` after `/api/v1/write/file/nquad` and it works fine:
```bash
curl -F "NQuadFile=@./data/testdata.nq" http://localhost:64210/api/v1/write/file/nquad
curl -H "Content-Type: application/n-quads" --data-binary "@./data/testdata.nq" http://localhost:64210/api/v2/delete
```
These `v1` and `v2` endpoint are compatible in sense that they can read and write the same data. But `/v1/write` and `/v1/delete` uses a hack to make JSON strings convertible to IRIs in some cases. Thus, you may see a difference when writing to these two endpoints.  

@lnshi Yes, v0.6.0 can already read/write/delete JSON-LD. All you need is to set `Content-Type: application/ld+json` for your request (only works for `/api/v2`).

Note that for `/api/v2/read` header is different: `Accept: application/ld+json`. GET request with query parameter `format=jsonld` will also work for this endpoint.

**Comment by dennwc** [comment]
@lnshi Oh, sorry, v0.6.0 does not support this yet. It's only in v0.6.1 (current master).

**Comment by lnshi** [comment]
@dennwc Thank u very much for the instructions and explanations, after getting this problem solved, I will move on to the JSON-LD support.

Verified, yea, if like this, with my own [test data](https://github.com/lnshi/node-cayley/blob/master/test/data/test_api_writeFile_purpose.nq), `/v2/delete` works fine, but either curl `-d` or `--data-binary` with `/v1/delete` will get below error:
```
{"error": "invalid character '\u003c' looking for beginning of value"}
```
But other than this, my point is: I don't think that writing in data with one N-Quads file then completely delete it with passing in the exactly same file is a common use case. Usually the common use case should be: I import bunch of N-Quads data with one N-Quads file, then in the future maybe I want to delete some N-Quads with passing in equivalent JSON data or JSON-LD data, and this is what I tried to do in the test case, and also it is where I encountered this `/v1/delete` and `/v2/delete` compatibilities problem.

Refer to [test case here: line 210](https://github.com/lnshi/node-cayley/blob/master/test/apis/callback.js), It writes in data with the N-Quads file, then try to delete all the wrote in N-Quads with passing in equivalent JSON object, here the problem comes: if I use `/v1/delete`, it works perfectly (all nodes get deleted properly in MongoDB), but if I switch to `/v2/delete`, then will get the below error(the response body):
```
{ error: 'delete </user/shortid/23TplPLs> -- <userId> -> 23TplPLs: quad does not exist' }
```
None N-Quad get deleted, the count of nodes is still 12.

So, in short words: I want u to help me to find out what difference between `v1/delete` and `/v2/delete` cause with the same test code base and data base, one works as expected, but one cannot work.

BTW, I also tried to quote the `false` with `"` in the passed in JSON object, still, `/v2/delete` reports same error.

And my cayley is running against `quay.io/cayleygraph/cayley:latest`, I just updated the container in the morning.
